src/Microsoft.Health.Dicom.Core/Modules/ServiceModule.cs

```diff
@@ -142,28 +133,25 @@ public void Load(IServiceCollection services)
                    .AsSelf()
                    .AsImplementedInterfaces();
 
-            if (_featureConfiguration.EnableExtendedQueryTags)
-            {
-                services.Add<ExtendedQueryTagEntryValidator>()
-                    .Singleton()
-                    .AsSelf()
-                    .AsImplementedInterfaces();
+            services.Add<ExtendedQueryTagEntryValidator>()
```

Comment:
Why are we always adding these services if they're functionality is dependent on the config? Is it to provide better errors to the user if they try to use a feature that's not enabled?

Reply:
if services are not created, customer would get internal server failure.